Source/towners.cpp

```diff
@@ -1157,17 +1141,18 @@ void __fastcall TalkToTowner(int p, int t)
 void __fastcall CowSFX(int pnum)
 {
 	if (CowPlaying == -1 || !effect_is_playing(CowPlaying)) {
-		if (sgdwCowClicks++ < 8) {
-			CowPlaying = (sgdwCowClicks == 4) + TSFX_COW1;
-		} else {
+		sgdwCowClicks++;
+		if (sgdwCowClicks >= 8) {
 			PlaySfxLoc(TSFX_COW1, plr[pnum].WorldX, plr[pnum].WorldY + 5);
 			sgdwCowClicks = 4;
 			CowPlaying = snSFX[sgnCowMsg][plr[pnum]._pClass]; /* snSFX is local */
-			if (sgnCowMsg++ >= 2)
+			sgnCowMsg++;
+			if (sgnCowMsg >= 3)
 				sgnCowMsg = 0;
+		} else {
+			CowPlaying = sgdwCowClicks == 4 ? 1 : 0;

```

Comment:
Probably not bin exact, but if it is, this looks cleaner:

```diff
- 			CowPlaying = sgdwCowClicks == 4 ? 1 : 0; 
- 			CowPlaying += TSFX_COW1; 
+ 			CowPlaying = sgdwCowClicks == 4 ? TSFX_COW1+1 : TSFX_COW1;
```

Reply:
CowPlaying = sgdwCowClicks == 4 ? TSFX_COW2 : TSFX_COW1;
;)

Reply:
Of cause :D Thanks, this has now been corrected :)